addons/l10n_ar/models/account_journal.py

```diff
@@ -112,16 +136,22 @@ def _check_afip_configurations(self):
                 _("You can not change the journal's configuration if it already has validated invoices") + ' ('
                 + ', '.join(invoices.mapped('journal_id').mapped('name')) + ')')
 
+    @api.constrains('l10n_ar_afip_pos_system')
+    def _check_afip_pos_system(self):
+        journals = self.filtered(
+            lambda x: x.l10n_ar_is_pos and x.type == 'purchase' and
+            self.l10n_ar_afip_pos_system not in ['II_IM', 'RLI_RLM', 'RAW_MAW'])
+        if journals:
+            raise ValidationError("\n".join([_(
+                "The pos system %s can not be used on a purchase journal (id %s)"
+                ) % (x.l10n_ar_afip_pos_system, x.id) for x in journals]))
+
     @api.constrains('l10n_ar_afip_pos_number')
     def _check_afip_pos_number(self):
-        to_review = self.filtered(
-            lambda x: x.type == 'sale' and x.l10n_latam_use_documents and
-            x.company_id.account_fiscal_country_id.code == "AR")
-
-        if to_review.filtered(lambda x: x.l10n_ar_afip_pos_number == 0):
+        if self.filtered(lambda x: x.l10n_ar_is_pos and x.l10n_ar_afip_pos_number == 0):
```

Comment:
```suggestion
        if self.filtered(lambda j: j.l10n_ar_is_pos and j.l10n_ar_afip_pos_number == 0):
```
I don't mind using a single letter in the filter, but I prefer it to be the first of what we are iterating on. (Journals here)